code/_expts/slotnick_etal_2001/de_StimCreate.m

```diff
@@ -19,70 +19,186 @@
 if (~exist('opt','var')),      opt      = {};     end;
 if (~iscell(opt)),             opt      = {opt};  end;
 
-% Create the input images.
-train.nInput = [68 50];
-train.X = zeros(prod(train.nInput), 0);
-train.XLAB = cell(20, 1); % Slotnick has 20 stimuli: 10 far/off, 5 near/off, 5 near/on
-train.TLAB = cell(20, 1);
-%train.XLAB = guru_csprintf('height=%.2f', num2cell(heights);
-counter = 0;
-heights = [];
-if (~strcmp(stimSet, 'blob-dot')) %default set of images
-    error('Invalid stimulus type. Choose one from: {''blob-dot''}');
+if (~strcmp(stimSet, 'blob-dot') && ~strcmp(stimSet, 'paired-squares')) %default set of images
```

Comment:
More concise to use `ismember`. You can also pass a cell array as the second argument to `strcmp`, and it will return a vector of ones and zeros (for any matches), and you can call check `~any` on that. `ismember` is the best, though.